apps/extension/src/state/tx-approval.ts

```diff
@@ -13,15 +23,114 @@ export interface TxApprovalSlice {
    * that everything be JSON-serializeable. So we'll store `Stringified`
    * representations of them instead.
    */
+  responder?: MessageResponder<TxApproval>;
   authorizeRequest?: Stringified<AuthorizeRequest>;
+  transactionView?: Stringified<TransactionView>;
+
+  acceptRequest: (
+    req: {
+      authorizeRequest: Jsonified<AuthorizeRequest>;
+      transactionView: Jsonified<TransactionView>;
+    },
+    responder: MessageResponder<TxApproval>,
+  ) => Promise<void>;
+
+  // these are generated by setTransactionView
   asSender?: Stringified<TransactionView>;
   asReceiver?: Stringified<TransactionView>;
   asPublic?: Stringified<TransactionView>;
   transactionClassification?: TransactionClassification;
-  // Holding the message responder function. Service worker will be "awaiting" the call of this.
-  responder?: MessageResponder<TxApproval>;
+
+  attitude?: boolean;
```

Comment:
Is this a normal name for a boolean representing user consent? To me, it seems like it'd be clearer to use a name like `isApproved` or something. But if `attitude` is a normal name for this concept that I'm not familiar with, LGTM :)